unified-runtime/include/ur_api.h

```diff
@@ -7431,6 +7431,11 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventSetCallback(
 ///////////////////////////////////////////////////////////////////////////////
 /// @brief Enqueue a command to execute a kernel
 ///
+/// @details
+///     - Adapters may perform validation on the number of arguments set to the
+///       kernel, but are not required to do so and may return
+///       `::UR_RESULT_SUCCESS` even for invalid invocations.
```

Comment:
I wonder if we should just say that passing incorrect kernel arguments is "undefined behavior" or something like that.

As shown in the test, if an adapter is unable to validate the arguments it's unlikely to be able to gracefully return success as well.

If we want to keep some level of argument validation, maybe it could be guarded behind a device property?

Reply:
Well it also segfaults on HIP, where we don't have a straightforward way of validating arguments either. So there's at least two targets where this doesn't really work.

And it seems to me that it defeats the purpose of making argument validation optional if we want targets to still do something specific when the arguments are wrong, if we could tell that the arguments are wrong then we'd just return the correct error code.

As it stands we can't tell if the arguments are wrong or not, so we just let the underlying API do whatever it can, which for CUDA seems to be to returning invalid value, and for HIP just segfaulting, if we want to fix that to return success, we pretty much circle back to having to validate the arguments in UR.